NYPL/research-catalog-indexer
# SCC-3580: Fix a few small indexing issues

EsItem:
 - Implement `EsItem.formatLiteral`
 - Ensure LCCNs are included in identifiers
 - Ensure ES model methods that return `null` are not included in to-json
 - Fix `idOclc` implementation
 - Fix `EsItem.aeonUrl` implementation

EsBib:
 - Fix bug relating holdings
 - Add `dateString`
 - Add `numCheckinCardItems`, `numItemDatesParsed`, `numItemVolumesParsed`, and `numItemsTotal`
 - Add " -- " joiner to `subjectLiteral` mapping

EsCheckinCardItem
 - Fix `formatLiteral`

EsHolding
 - Fix `holdingStatement`
 - Change `location.id` to `location.code`
 - Ensure `checkinBoxes` is sorted by `position`

Also 
 - Expands test coverage in a few areas
 - Alphabetizes `EsItem` and `EsBib` methods (sorry)
 - Fixes compare-with-indexed script, broken by recent changes to build-es-document
 - Adds item date parsing (and wires it up in general prefetch)
 - ToJson should not try to index `null` values
 - Passing "true" to `compare-with-indexed --verbose` should be optional
